docs/en/managing-data/updates.md

```diff
@@ -76,6 +81,51 @@ Note that for lightweight updates, a mutation is still used to update the data;
 
 Read more about [lightweight updates](/en/guides/developer/lightweight-update).
 
+## Collapsing Merge Tree
+
+Stemming from the idea that updates are expensive but inserts can be leveraged to perform updates,
+the [`CollapsingMergeTree`](/en/engines/table-engines/mergetree-family/collapsingmergetree) table engine 
+can be used together with a `sign` column as a way to tell ClickHouse to update a specific row by collapsing (deleting)
+a pair of rows with sign `1` and `-1`. 
+If `-1` is inserted for the `sign` column, the whole row will be deleted. 
+If `1` is inserted for the `sign` column, ClickHouse will keep the row. 
+Rows to update are identified based on the sorting key used in the `ORDER BY ()` statement when creating the table.
+
+```sql
+CREATE TABLE UAct
+(
+    UserID UInt64,
+    PageViews UInt8,
+    Duration UInt8,
+    Sign Int8 -- A special column used with the CollapsingMergeTree table engine
+)
+ENGINE = CollapsingMergeTree(Sign)
+ORDER BY UserID
+
+INSERT INTO UAct VALUES (4324182021466249494, 5, 146, 1)
+INSERT INTO UAct VALUES (4324182021466249494, 5, 146, -1) -- sign = -1 signals to update the state of this row
+INSERT INTO UAct VALUES (4324182021466249494, 6, 185, 1) -- the row is replaced with the new state
+
+SELECT
+    UserID,
+    sum(PageViews * Sign) AS PageViews,
+    sum(Duration * Sign) AS Duration
+FROM UAct
+GROUP BY UserID
+HAVING sum(Sign) > 0
+
+┌──────────────UserID─┬─PageViews─┬─Duration─┐
+│ 4324182021466249494 │         6 │      185 │
+└─────────────────────┴───────────┴──────────┘
+```
+
+:::note
```

Comment:
They key thing here is it requires the user to maintain state client side - while its the most efficient for clickhouse, this complex at scale. We should state this.

Reply:
Update pushed